src/Storages/ObjectStorage/StorageObjectStorage.h

```diff
@@ -167,8 +167,7 @@ class StorageObjectStorage::Configuration
     using Path = std::string;
     using Paths = std::vector<Path>;
 
-    static void initialize(
-        Configuration & configuration,
+    virtual void initialize(
```

Comment:
Is there a strong reason on why you made this a member instead of a static method?

I don't have a preference, perhaps a member is indeed better, but makes the review process way harder. There are a few changes in this PR, and the other ones I reviewed, that seem kind of unrelated.

I think it would be best if you did the refactorings in different PRs and kept changes to a minimum, that would make the review process much easier

Reply:
I need to override it in StorageIcebergConfiguration, and virtual member can't be static.